hw/dv/tools/ralgen/ralgen.py

```diff
@@ -18,9 +19,22 @@
 except ImportError:
     from yaml import SafeLoader as YamlLoader
 
-# Repo root is 4 levels up. Note that this will require an update if the path to
-# this tool is changed.
-REPO_ROOT = "../../../.."
+
+# Get the REPO_TOP directory path. This is used to construct full paths below.
```

Comment:
looks great! 
i wish there were a common util library area where you could shove this into.. i have to imagine this will come in handy elsewhere too.

Reply:
yeah this is basically, copy-pasted from dvsim.py. I created #13073 to track this.